### PR TITLE
Support Django's save method `update_fields` kwarg

### PR DESCRIPTION
**Pull request by AbdullahKady**
Solves #243 

This has been an irritating issue that I've been facing, the [issue](https://github.com/jazzband/django-auditlog/issues/243) seems to describe it quite well.
Please let me know if you have any comments :)

### CODE REVIEW

**Comment by AbdullahKady** [comment]
@hramezani @Linkid can I get an update regarding this please? I cannot request a review on the PR.
Just in case if there is anything blocking from my side, or if there is a decision that this bug is not relevant.

**Comment by hramezani** [comment]
Let's wait for @Linkid pr https://github.com/jazzband/django-auditlog/pull/339 and then the CI will be green.

@Linkid Could you please take a look meanwhile?

File: auditlog_tests/tests.py [hunk 1]
```diff
@@ -81,6 +81,49 @@ def test_update(self):
...
+        self.assertJSONEqual(
+            obj.history.get(action=LogEntry.Action.UPDATE).changes,
+            '{"boolean": ["False", "True"]}',
+            msg="Object modifications that are not saved to DB are not logged when using the `update_fields`.",
+        )
+
+    def test_django_update_fields_edge_cases(self):
+        """
+        The test ensures that if Django's `update_fields` behavior was ever changed for special values (None, []),
+        the package should too. https://docs.djangoproject.com/en/models/instances/#specifying-which-fields-to-save
```
Comment (Linkid) [inline auditlog_tests/tests.py:102]: The link is not working. You can use this one instead: https://docs.djangoproject.com/en/3.2/ref/models/instances/#specifying-which-fields-to-save (3.2 is the latest LTS).

File: auditlog_tests/tests.py [hunk 2]
```diff
@@ -81,6 +81,49 @@ def test_update(self):
...
+        obj.save(update_fields=[])
+
+        self.assertTrue(
+            obj.history.filter(action=LogEntry.Action.UPDATE).count() == 0,
+            msg="There is no log entries created",
+        )
+        obj.refresh_from_db()
+        self.assertFalse(obj.boolean)  # Change didn't persist in DB as expected.
+
+        # Passing `None`` should save both fields according to Django.
```
Comment (Linkid) [inline auditlog_tests/tests.py:117]: Typo: ``` `None`` ``` → `None`

**Review by Linkid — CHANGES_REQUESTED** [review]
Nice ! Some minor changes to do and it will be great ! Thank you.

**Review by Linkid — APPROVED** [review]
LGTM

**Review by hramezani — APPROVED** [review]
Thanks @AbdullahKady

**Pull request merged**